sql/src/test/java/io/crate/integrationtests/GroupByAggregateTest.java

```diff
@@ -1159,4 +1159,30 @@ public void testFilterByScore() throws Exception {
         assertEquals(2L, response.rowCount());
     }
 
+    @Test
+    public void testDistinctWithGroupBy() throws Exception {
+        execute("create table t (a integer, b integer)");
+        ensureYellow();
+        execute("insert into t(a, b) values(1,1), (1,2), (1,3), (2,1), (2,2), (2,3), (2,4), (3,1), (3,2)");
+        execute("refresh table t");
+
+        // Simple select
+        execute("select DISTINCT max(a), min(a) from t group by b order by 2, 1");
```

Comment:
maybe create separate tests? Could also consider using the unnest table function to avoid the create table/insert overhead